src/chrome/internal/services/sourceToScriptMapper.ts

```diff
@@ -21,8 +22,8 @@ export class SourceToScriptMapper {
         this.doesTargetSupportColumnBreakpointsCached = this._breakpointFeaturesSupport.supportsColumnBreakpoints;
     }
 
-    public async mapBPRecipe(bpRecipe: BPRecipeInLoadedSource<ConditionalPause | AlwaysPause>): Promise<BPRecipeInScript[]> {
-        const tokensInManyScripts = bpRecipe.location.tokensWhenMappedToScript();
+    public async mapBPRecipe(bpRecipe: BPRecipeInLoadedSource<ConditionalPause | AlwaysPause>, onlyKeepIfScript: (script: IScript) => boolean = () => true): Promise<BPRecipeInScript[]> {
+        const tokensInManyScripts = bpRecipe.location.tokensWhenMappedToScript().filter(tokens => onlyKeepIfScript(tokens.script));
```

Comment:
Just to clarify on what this filter is doing. It looks like this is accounting for the case where we have a resource that could map to multiple script sources, and we only care about the script that was recently parsed, correct? And the only way, then, to check if the breakpoint is in the script we just parsed is to resolve all of them associated with that resource?

Do we have to do the mapping resolve every time we process any breakpoint associated with that resource?